File: Algorithms/Ones2/ones2.py
mod = 1000000007
def step(a, n):
    res = 1
    while(n > 0):
        if n%2==1:
            res = (res%mod*a%mod)%mod
        a = (a%mod*a%mod)%mod
        n>>=1
    return res

def C(n, k):
    if k==0:
        return 1
    if k>n:
        return 0
    div1 = 1
    for i in range(0, k):
        div1 = (div1%mod*(n-i)%mod)%mod
    div2 = 1
    for i in range(1, k+1):
        div2 = (div2%mod*i%mod)%mod
    return int(div1%mod * step(div2, mod-2))%mod

n, k = map(int, input().split())
if k > n-k:
    print(C(n,n-k))
else:
    print(C(n,k))

File: Algorithms/Ones2/ones2.cpp
#include <iostream>
#include <vector>

unsigned long long step(int a, unsigned int b) {
    if (b == 0) 
        return 1;
    int c = 1;
    while (b != 0) {
        if (b % 2 == 1) {
            c = c * a;
            b--;
        }
        a = a * a;
        b = b / 2;
    }
    return c;
}

const long long mod = 1000000007;

unsigned long long C(int n, int k) {
    return 0;
}

using namespace std;

int main() {
    int N, K;
    cin >> N >> K;
    if (K > N - K)
        cout << C(N, N - K) << "\n";
    else
        cout << C(N, K) << "\n";
}